example/src/main/java/io/netty/example/tunechoserver/TunEchoServer.java

```diff
@@ -0,0 +1,452 @@
+        if (ipHeader.ipVersion == InetUtils.IP_VERSION_4) {
+
+            // Read the portion of the packet containing the new ICMP checksum bytes.
+            byte[] newICMPChecksumBytes = new byte[2];
+            packetBuf.getBytes(ipHeader.headerLength + InetUtils.ICMP_HEADER_OFFSET_CHECKSUM, newICMPChecksumBytes);
+
+            // Compute an updated IP checksum.
+            int newIPChecksum = InetUtils.updateInetChecksum(oldICMPChecksumBytes, 0, newICMPChecksumBytes,
+                    0, 2, ipHeader.checksum);
+            newIPChecksum = InetUtils.updateInetChecksum(oldTypeAndCodeBytes, 0, newTypeAndCode, 0, 2, newIPChecksum);
+
+            // Update the checksum field in the IPv4 header.
+            packetBuf.setShort(InetUtils.IPV4_HEADER_OFFSET_CHECKSUM, newIPChecksum);
+        }
+    }
+
+    private static boolean readProps() {
+
+        _tunDeviceName = System.getProperty("tun-device-name", "tun-echo-server");
+
+        try {
+            _threadPoolSize = Integer.parseInt(System.getProperty("thread-pool-size", "0"));
+            if (_threadPoolSize < 0) {
+                throw new IllegalArgumentException();
+            }
+        } catch (Exception ex) {
+            System.err.println("Invalid value specified for thread-pool-size property: " +
+                    System.getProperty("thread-pool-size"));
+            return false;
+        }
+
+        try {
+            _readBufSize = Integer.parseInt(System.getProperty("read-buf-size", "2048"));
+            if (_readBufSize < 1) {
+                throw new IllegalArgumentException();
+            }
+        } catch (Exception ex) {
+            System.err.println("Invalid value specified for read-buf-size property: " +
+                    System.getProperty("read-buf-size"));
+            return false;
+        }
+
+        if (Boolean.parseBoolean(System.getProperty("use-unpooled", "false"))) {
+            _bufAllocator = UnpooledByteBufAllocator.DEFAULT;
+        }
+
+        _forceGC = Boolean.parseBoolean(System.getProperty("force-gc", "false"));
+
+        _debug = Boolean.parseBoolean(System.getProperty("debug", "true"));
+
+        return true;
+    }
+
+    public static void main(String[] args) {
+
+        if (!readProps()) {
+            return;
+        }
+
+        System.out.println("TunEchoServer starting");
+
+        try {
+            _logger = Logger.getLogger("");
+            _logger.setLevel(_debug ? Level.FINEST : Level.SEVERE);
+
+            if (_threadPoolSize > 0) {
+                _threadPool = (ThreadPoolExecutor) Executors.newFixedThreadPool(_threadPoolSize);
+            }
+
+            final TunEchoServer server = new TunEchoServer();
+
+            EventLoopGroup group = new EpollEventLoopGroup(1);
+
+            ChannelFactory<Channel> channelFactory = new ChannelFactory<Channel>() {
+                @Override
+                public Channel newChannel() {
+                    try {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested try block into a separate method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1141)